### ISSUE TITLE
Getting Instance is already consumed when reading two subsequent values from Input

### ISSUE THREAD

**Issue by altavir** [Kotlin/kotlinx-io#23]
on 0.1.1 I am getting `java.lang.IllegalStateException: Instance is already consumed` when trying read fixed length string after int from input. It works just fine in 0.1.0.

The full stack trace is following:
```
java.lang.IllegalStateException: Instance is already consumed

	at kotlinx.io.pool.SingleInstancePool.borrow(Pool.kt:73)
	at kotlinx.io.core.ByteReadPacketBase.fixGapAfterReadFallbackUnreserved(Packet.kt:662)
	at kotlinx.io.core.ByteReadPacketBase.fixGapAfterReadFallback(Packet.kt:644)
	at kotlinx.io.core.ByteReadPacketBase.fixGapAfterRead(Packet.kt:618)
	at kotlinx.io.core.internal.UnsafeKt.completeReadHead(Unsafe.kt:66)
	at kotlinx.io.charsets.CharsetJVMKt.decode(CharsetJVM.kt:555)
	at kotlinx.io.charsets.EncodingKt.decode(Encoding.kt:104)
	at kotlinx.io.core.StringsKt.readText(Strings.kt:339)
	at kotlinx.io.core.StringsKt.readText$default(Strings.kt:338)
```

I can give reference to the test if it would help.

**Comment by cy6erGn0m** [comment]
I can't reproduce it. Could you please provide a reference?

**Comment by altavir** [comment]
[This test](https://github.com/altavir/dataforge-core/blob/5eb1ffb38540c4a3fa580dc4acf27e2d419ff016/dataforge-meta-io/src/commonTest/kotlin/hep/dataforge/meta/io/MetaFormatTest.kt#L9) seems to be working on 0.1.0, but does not work on 0.1.1.

**Comment by cy6erGn0m** [comment]
Ah I see now

**Comment by cy6erGn0m** [comment]
Fixed in 0.1.2-dev-1
